front/lib/auth.ts

```diff
@@ -631,6 +631,9 @@ export class Authenticator {
   }
 
   hasPermission(acls: ACLType[], permission: Permission): boolean {
+    if (this.isAdmin()) {
+      return true;
```

Comment:
This should not be true.

An admin can administrate groups but cannot access the data. This is an important distinction.

Reply:
There's an issue with the fact we are using the same permission for viewing a vault and being able to read the data, we should probably have 2 different permissions. An admin can view and administrate connections , but may not be able to read data ? 
Another problem is that a vault, by default, may not be readable by anybody - should we add the vault creator inside the group ? What if all members of the vault are removed ?

Reply:
We keep permissions for viewing/reading/writing in the vault, fine if the vault is empty.

And admins can view vaults and administrate them (aka list and edit data sources and views content) but they cannot read write into data sources content or view content.

Basically an admin can create a CEO vault and add a view to it with stuff from the CEO shared drive. But at no moment in time should they be able to read from the vault data source or data source views content.

Reply:
We keep permissions for viewing/reading/writing in the vault, fine if the vault is empty.

And admins can view vaults and administrate them (aka list and edit data sources and views content) but they cannot read write into data sources content or view content.

Basically an admin can create a CEO vault and add a view to it with stuff from the CEO shared drive. But at no moment in time should they be able to read from the vault data source or data source views content.```

```

Reply:
So we def don't want the creator to be in the vault as per the scenario above.

We might want a new admin permission on top of read / write and some operations will be read or admin, or, write or admin.

And admins will always have this admin permission.

And some operations like reading the content will be read only 👍

We can also simply test if the user is admin instead of checking the admin permission but maybe having that admin permission will make the code cleaner... Both work.

Makes sense?

Reply:
Ok, so I won't change anything into authenticator - not giving admin more permissions - but allow admin to see and administrate vaults on which he has no permissions. No need to add them to the vault then (although they can do it manually, since they can administer the vaults). 

Non admin users will only see the vaults on which they have access.

Reply:
Sounds good!

Of course admins can add themselves but since all admins operations will be audit logged they won't be able to do so without it being visible 👍